### PR TITLE
Hypergraph refactor

### PR DESCRIPTION
**Pull request by tomichels**
## Summary
- Replaces holoviews DynamicMap-based hypergraph rendering with static matplotlib, dropping the bokeh dependency.
- Adds pattern_size, must_involve_neuron filters and node_size/node_color/node_linewidth/show_ids styling options to plot_patterns_hypergraph.
- Adds unit tests for Hypergraph.
- CI: doc build now fails on Sphinx warnings instead of swallowing them

### CODE REVIEW

File: viziphant/patterns.py [hunk 1]
```diff
@@ -429,11 +430,22 @@ def plot_patterns_hypergraph(patterns, num_neurons=None):
...
         integer is passed, it identifies the total number of recorded neurons
         including non-pattern neurons to be additionally shown in the graph.
         Default: None
+    must_involve_neuron (optional) : int
+        Highlight pattern which includes neuron x
+    node_color (optional) : String
+        change the color of the nodes
+
+    node_linewidth (optional) : int
+            change the line width of the nodes
```
Comment (mdenker) [inline viziphant/patterns.py:449]: I think the order of the individual options seems a bit random. I think users will expect first the options that control *what* is shown (e.g., number of neurons), then *how* its shown (e.g., node color).

File: viziphant/patterns.py [hunk 2]
```diff
@@ -461,8 +473,7 @@ def plot_patterns_hypergraph(patterns, num_neurons=None):
         bst.rescale('ms')
         patterns = cell_assembly_detection(bst, max_lag=2)
 
-        fig = viziphant.patterns.plot_patterns_hypergraph(patterns)
-        plt.show()
+        viziphant.patterns.plot_patterns_hypergraph(patterns)
```
Comment (mdenker) [inline viziphant/patterns.py:482]: What was the reason to remove the `fig = ` and `plt.show()` parts?

File: viziphant/patterns_src/hypergraph.py [hunk 3]
```diff
@@ -104,15 +104,23 @@ def complete_and_star_associated_graph(self):
         edges = []
         weights = []
         graph_vertices = list(self.vertices.copy())
+        if isinstance(self.vertices[0], int):
+            max_vertex = max(max(hyperedge) for hyperedge in self.hyperedges)
+        else:
+            max_vertex = 0
+        if isinstance(max_vertex, str):
+            max_vertex = 0
```
Comment (mdenker) [inline viziphant/patterns_src/hypergraph.py:112]: Can you include comments on the logic of these if statements? Why int and str?

Comment (mdenker) [reply viziphant/patterns_src/hypergraph.py:112]: I assume you need this to get unique IDs for the pseudo vertecies?

File: viziphant/patterns_src/view.py [hunk 4]
```diff
@@ -44,6 +44,15 @@ def __init__(self, hypergraphs, title=None):
...
             Each hypergraph should contain data of one data set.
+
+        node_size (optional) : int
+            Size of the nodes in the Hypergraphs
+        
+        node_color (optional) : String
+            change the color of the nodes
+
+        node_linewidth (optional) : int
+            change the line width of the nodes
```
Comment (mdenker) [inline viziphant/patterns_src/view.py:64]:
```suggestion
        node_size (optional) : int
            Size of the nodes in the Hypergraphs
        node_color (optional) : String
            change the color of the nodes
        node_linewidth (optional) : int
            change the line width of the nodes
```

File: viziphant/test/test_hypergraph.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+        hypergraph = Hypergraph(
+            hyperedges, vertices, vertex_labels, weights, positions, repulse,
+        )
+
+        graph = hypergraph.complete_and_star_associated_graph()
+        self.assertLessEqual(
+            nx.dijkstra_path_length(graph, 1, 2), nx.dijkstra_path_length(graph, 1, 4)
+        )
+
+    # Two Hypergraphs with the same data should be equal
```
Comment (mdenker) [inline viziphant/test/test_hypergraph.py:324]: I think all unit tests should have such a sentence as a comment that briefly outlines the idea of the test.

File: viziphant/test/test_hypergraph.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+        # self.patterns = elephant.spade.spade(
+        #    spiketrains=self.spiketrains,
+        #    binsize=1 * pq.ms,
+        #    winlen=1,
+        #    min_spikes=2,
+        #    n_surr=100,
+        #    dither=5 * pq.ms,
+        #    psr_param=[0, 0, 0],
+        #    output_format="patterns",
+        # )["patterns"]
```
Comment (mdenker) [inline viziphant/test/test_hypergraph.py:37]: Why are these lines commented out -- forgotten or kept for a potential later unit test? In the latter case, please introduce a TODO comment. Otherwise, in the former case, please remove this code.

**Review by mdenker — CHANGES_REQUESTED** [review]

**Comment by CozySocksAlways** [comment]
While working on PR #83 , I encountered exception raises on sphinx doc build job in the CI. But this is silently ignored by Sphinx and the actual build succeeds, to prevent this, a stricter doc build check is needed, but this cannot be implemented in PR #83 as it needs this current PR #77 to be merged, as `plot_patterns_hypergraph` method also fails in the doc strings now.
TLDR; add stricter doc build to CI when merging this PR, specifically switch to `make html SPHINXOPTS="-W --keep-going"` in the CI.
